cmd/arena/commands/diagnose.go

```diff
@@ -0,0 +1,324 @@
+			fmt.Printf("Collecting relevant logs of job %s in %v, please wait...\n", jobName, path)
+			diagnoseJob(client, jobResources, diagnoseArgs.jobType, jobName, path)
+			fmt.Println("Done.")
+		},
+	}
+	command.Flags().StringVarP(&diagnoseArgs.outputDir, "outputDir", "o", "", "The output direction of the collected logs.")
+	command.Flags().StringVarP(&diagnoseArgs.jobType, "type", "t", "", "The type of the selected job.")
+	if err := command.MarkFlagRequired("type"); err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	return command
+}
+
+func diagnoseJob(client *kubernetes.Clientset, jobResources TrainingJobResources, jobType, jobName, dirPath string) {
+	timestamp, dirName := prepareFolder(jobType, jobName, dirPath)
+	// 0. prepare environment
+	kubePath, err := exec.LookPath(KUBECTL_BIN)
+	if err != nil {
+		log.Errorf("Failed to get the kubectl path due to %v", err)
+		return
+	}
+	if types.KubeConfig != "" {
+		env = append(env, fmt.Sprintf("KUBECONFIG=%s", types.KubeConfig))
+	}
+
+	// create description sub folder
+	descDirName := filepath.Join(dirName, "description")
+	_ = createDir(descDirName)
+
+	// 1. describe job
+	describeJob(kubePath, dirName, jobType, jobName)
+
+	// 2. describe sts
+	describeSts(jobResources, kubePath, descDirName)
+
+	// 3. describe batch job
+	describeBatchJob(jobResources, kubePath, descDirName)
+
+	// 4. describe pod
+	describePod(jobResources, kubePath, descDirName)
+
+	// 5. describe the corresponding nodes
+	describeNode(client, jobResources, kubePath, descDirName)
+
+	// create logs sub folder
+	logDirName := filepath.Join(dirName, "logs")
+	_ = createDir(logDirName)
+
+	// 6. pod logs
+	logsPod(jobResources, kubePath, logDirName)
+
+	// 7. job-operator logs
+	logsOperatorPod(jobResources, kubePath, logDirName)
+
+	// 8. get events
+	getEvents(kubePath, dirName, timestamp)
+}
+
+// create the dest prepare folder
+func prepareFolder(jobType, jobName, dirPath string) (string, string) {
+	timestamp := strconv.FormatInt(time.Now().Unix(), 10)
+	dirName := filepath.Join(dirPath, fmt.Sprintf("diagnose-%s-%s-%s", jobType, jobName, timestamp))
+	fmt.Printf("The diagnose folder is %s.\n", dirName)
+
+	if err := createDir(dirName); err != nil {
+		log.Errorf("Failed to create the log folder due to %v", err)
+		os.Exit(1)
+	}
+	return timestamp, dirName
+}
+
+func describeJob(kubePath, dirName, jobType, jobName string) {
+	var job string
+	switch jobType {
+	case "tfjob":
+		job = "tfjobs.kubeflow.org"
+	case "mpijob":
+		job = "mpijobs.kubeflow.org"
+	case "sparkjob":
+		job = "sparkapplications.sparkoperator.k8s.io"
+	case "volcanojob":
+		job = "jobs.batch.volcano.sh"
+	default:
+		log.Fatalf("Unsupported job type: %s", jobType)
+	}
+	describeArgs := []string{KUBE_DESCRIBE, "-n", namespace, job, jobName}
+	saveKubectlCmdResult(kubePath, describeArgs, dirName, jobName)
+}
+
+func describeSts(jobResources TrainingJobResources, kubePath, descDirName string) {
```

Comment:
Could you please move these methods into `github.com/kubeflow/arena/pkg/util/kubectl` ? Thanks.